Src/Idioms/EqualityAssertion.cs

```diff
@@ -0,0 +1,364 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.Idioms
+{
+	/// <summary>
+	/// Represent an assertion which verifies that an equality method returns the correct value.
+	/// </summary>
+	public class EqualityAssertion : IdiomaticAssertion
+	{
+		private readonly Func<object, object, bool> comparer;
+		private IFixture fixture;
+
+		/// <summary>
+		/// Creates a new instance of EqualityAssertion which verified the Object.Equals method
+		/// </summary>
+		/// <param name="fixture">A composer which can create instances required to implement the idiomatic unit test.</param>
+		public EqualityAssertion(IFixture fixture)
+			: this(fixture, (x, y) => x.Equals(y))
```

Comment:
You are right, I thought there only was a generic version of IEqualityComparer, thanks :)